homeassistant/components/light/mqtt.py

```diff
@@ -0,0 +1,171 @@
+"""
+homeassistant.components.light.mqtt
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Allows to configure a MQTT light.
+"""
+import logging
+import homeassistant.components.mqtt as mqtt
+from homeassistant.components.light import (Light,
+                                            ATTR_BRIGHTNESS, ATTR_RGB_COLOR)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "MQTT Light"
+DEFAULT_QOS = 0
+DEFAULT_PAYLOAD_ON = "on"
+DEFAULT_PAYLOAD_OFF = "off"
+DEFAULT_RGB = [255, 255, 255]
+DEFAULT_RGB_PATTERN = "%d,%d,%d"
+DEFAULT_BRIGHTNESS = 120
+
+DEFAULT_STATE_TOPIC = "homeassistant/light/state"
+DEFAULT_COMMAND_TOPIC = "homeassistant/light/switch"
+
+DEFAULT_STATE_BRIGHTNESS = "homeassistant/light/brightness/state"
+DEFAULT_COMMAND_BRIGHTNESS = "homeassistant/light/brightness/set"
+
+DEFAULT_STATE_RGB = "homeassistant/light/rgb/state"
+DEFAULT_COMMAND_RGB = "homeassistant/light/rgb/set"
+
+DEPENDENCIES = ['mqtt']
+
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Add MQTT Light. """
+
+    if config.get('command_topic') is None:
+        _LOGGER.error("Missing required variable: command_topic")
+        return False
+
+    add_devices_callback([MqttLight(
+        hass,
+        config.get('name', DEFAULT_NAME),
+        config.get('state_topic', DEFAULT_STATE_TOPIC),
+        config.get('command_topic', DEFAULT_COMMAND_TOPIC),
+        config.get('brightness_state_topic', DEFAULT_STATE_BRIGHTNESS),
+        config.get('brightness_command_topic', DEFAULT_COMMAND_BRIGHTNESS),
+        config.get('rgb_state_topic', DEFAULT_STATE_RGB),
+        config.get('rgb_command_topic', DEFAULT_COMMAND_RGB),
+        config.get('rgb', DEFAULT_RGB),
+        config.get('qos', DEFAULT_QOS),
+        config.get('payload_on', DEFAULT_PAYLOAD_ON),
+        config.get('payload_off', DEFAULT_PAYLOAD_OFF),
+        config.get('brightness', DEFAULT_BRIGHTNESS))])
```

Comment:
Not all MQTT lights will have support to set the color or the brightness. The light should be able to function as just a on/off switch.